os/board/common/crashdump.c

```diff
@@ -129,6 +133,81 @@ static int get_userfs_address_size(uint32_t *userfs_start, size_t *userfs_size)
 	return 0;
 }
 
+static int get_ext_userfs_part(size_t *size, int *index)
+{
+#if defined(CONFIG_SECOND_FLASH_PART_NAME) && defined(CONFIG_SECOND_FLASH_PART_SIZE)
+	int i = 0;
+	int c = 0;
+	int x = 0;
+	size_t ext_userfs_size;
+	char second_flash_part_list[] = CONFIG_SECOND_FLASH_PART_NAME;
+	char second_flash_part_sizes[] = CONFIG_SECOND_FLASH_PART_SIZE;
+	char tmp_part_name[USERFS_PART_NAME_LEN + 1] = { '0' };
+
+	while (second_flash_part_list[i] != '\0') {
```

Comment:
some of this is duplicated with code in partitons.c , I guess...
so can't we reuse it if we add signature of those functions to partition.h ??

Reply:
The common part of code is one function as follows: -
```
static void move_to_next_part(const char **par)
{
                /* Move to next part information. */
                while (*(*par)++ != ',');
}
```
which parses the partition names/list till the next comma is encountered.

This can be made to fit with the requirement here.
However in addition to the new function `get_ext_userfs_part`, another pre-existing function `get_userfs_address_size` will make use of this.

So, I will make the changes for both and optimize in an upcoming commit.
Thankyou very much for pointing this out.